exasol/error/_error.py

```diff
@@ -0,0 +1,85 @@
+import warnings
+from collections.abc import Iterable, Mapping
+from dataclasses import dataclass
+from typing import Dict, List, Union
+
+with warnings.catch_warnings():
+    warnings.simplefilter("ignore")
+    from exasol_error_reporting_python import exa_error
+
+
+@dataclass(frozen=True)
+class Parameter:
+    value: str
+    description: Union[None, str]
+
+
+class Error:
+    def __init__(
+        self,
+        name: str,
+        message: str,
+        mitigations: Union[str, Iterable[str]],
+        parameters: Dict[str, Union[str, Parameter]],
+    ):
+        def build_error(code, msg, mitigations, params):
```

Comment:
a bit matter of taste, does this function really help to make it more readable and if should we add type hints to the parameter

Reply:
I don't think this should/can be done once the inner function is moved outside of the constructor. This may happen in the future when error reporting for the package itself will be implemented.